packages/key-management/src/types.ts

```diff
@@ -175,9 +177,11 @@ export interface KeyAgent {
 
 export type AsyncKeyAgent = Pick<KeyAgent, 'deriveAddress' | 'derivePublicKey' | 'signBlob' | 'signTransaction'> & {
   knownAddresses$: Observable<GroupedAddress[]>;
+  getKeyAgent(): Promise<KeyAgent>;
```

Comment:
That won't work with our messaging implementation, because returned object has both async methods and sync properties - not supported. Seems like we need to refactor `deriveAddress` to support deriving without mutating the key agent, probably add an argument like `pure=false`
```suggestion
```